File: Frontend/src/app/models/pregunta.ts
import { Respuesta } from "./respuesta";
import { Usuario } from "./usuario";

export class Pregunta {

    public id: number;
    public createdAt: Date;
    public pregunta: string;
    public descripcion: string;
    public aprobado: boolean;

    public respuestas: Respuesta[];
    public usuario: Usuario;

    public constructor(json?: any) {

        Object.assign(this, json);

        if (this.createdAt) { this.createdAt = new Date(this.createdAt) };
        if (this.respuestas) { this.respuestas = this.respuestas.map(respuesta => new Respuesta(respuesta)) };
        if (this.usuario) { this.usuario = new Usuario(this.usuario) };

    }

    public getJSON(): any {
        return {
            id: this.id,
            createdAt: this.createdAt,
            pregunta: this.pregunta,
            descripcion: this.descripcion,
            aprobado: this.aprobado
        };
    }

}

File: Backend/src/main/java/aqct/backend/model/Pregunta.java
package aqct.backend.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import java.io.Serializable;
import java.sql.Timestamp;
import java.util.List;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import org.hibernate.annotations.CreationTimestamp;

@Entity
@Table(name = "pregunta")
public class Pregunta implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @Column(name = "pregunta")
    private String pregunta;
    
    @Column(name = "descripcion")
    private String descripcion;
    
    @Column(name = "aprobado")
    private boolean aprobado;
    
    @Column(name = "created_at")
    @CreationTimestamp
    private Timestamp createdAt;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "usuario_id")
    @JsonManagedReference
    private Usuario usuario;
    
    @OneToMany(fetch = FetchType.LAZY)
    @JoinColumn(name = "pregunta_id")
    @JsonManagedReference
    private List<Respuesta> respuestas;
    
    public Pregunta() {
        
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getPregunta() {
        return pregunta;
    }

    public void setPregunta(String pregunta) {
        this.pregunta = pregunta;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public boolean isAprobado() {
        return aprobado;
    }

    public void setAprobado(boolean aprobado) {
        this.aprobado = aprobado;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }
    
}
